modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/ComponentMessageProcessor.java

```diff
@@ -503,6 +504,10 @@ public void doDispose() {
   }
 
   protected ExecutionMediator createExecutionMediator() {
+    if (executionMediatorFactory != null) {
```

Comment:
remove this if. The factory should default to a supplier with the default behavior.